pushmsixscript/src/pushmsixscript/microsoft_store.py

```diff
@@ -0,0 +1,151 @@
+import logging
+import time
+
+import requests
+from scriptworker_client.exceptions import TaskVerificationError
+
+from pushmsixscript import task
+
+log = logging.getLogger(__name__)
+
+# When committing a new submission, poll for completion, with
+# this many attempts, waiting this long between attempts.
+COMMIT_POLL_MAX_ATTEMPTS = 10
+COMMIT_POLL_WAIT_SECONDS = 30
+
+
+def push(config, msix_file_path, channel):
+    """Publishes a msix onto a given channel.
+
+    This function performs all the network actions to ensure `msix_file_path` is published on
+    `channel`. If `channel` is not whitelisted to contact the Microsoft Store, then it early
+    returns (this allows staging tasks to not contact the Microsoft Store at all).
+
+    Args:
+        config (config): the scriptworker configuration.
+        msix_file_path (str): The full path to the msix file to upload.
+        channel (str): The Microsoft Store channel.
+    """
```

Comment:
Elsewhere we use Google-style docstrings https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html . This looks good!

If this goes into a ReadTheDocs type generated code doc, I've found that:
- I tend to have to use double backticks because ReStructuredText:
```
   foo bar ``foobar``
```
- Best practice says we'd have `Returns:` and `Raises:` sections like `Args:`.

However, our docstring usage and style is pretty mixed in scriptworker-scripts, so I'm pretty open to just having things documented. Perhaps someday we'll lint docstrings and generate RtD docs from them.